src/linearization.jl

```diff
@@ -565,6 +565,51 @@ function linearize_symbolic(sys::AbstractSystem, inputs,
     (; A, B, C, D, f_x, f_z, g_x, g_z, f_u, g_u, h_x, h_z, h_u), sys
 end
 
+struct IONotFoundError <: Exception
+    variant::String
+    sysname::Symbol
+    not_found::Any
+end
+
+function Base.showerror(io::IO, err::IONotFoundError)
+    println(io,
+        "The following $(err.variant) provided to `mtkcompile` were not found in the system:")
+    maybe_namespace_issue = false
+    for var in err.not_found
+        println(io, "  ", var)
+        if hasname(var) && startswith(string(getname(var)), string(err.sysname))
+            maybe_namespace_issue = true
+        end
+    end
+    if maybe_namespace_issue
+        println(io, """
+        Some of the missing variables are namespaced with the name of the system \
+        `$(err.sysname)` passed to `mtkcompile`. This may be indicative of a namespacing \
+        issue. `mtkcompile` requires that the $(err.variant) provided are not namespaced \
+        with the name of the root system. This issue can occur when using `getproperty` \
+        to access the variables passed as $(err.variant). For example:
+
+        ```julia
+        @named sys = MyModel()
+        inputs = [sys.input_var]
+        mtkcompile(sys; inputs)
+        ```
+
+        Here, `mtkcompile` expects the input to be named `input_var`, but since `sys`
+        performs namespacing, it will be named `sys$(NAMESPACE_SEPARATOR)input_var`. To \
+        fix this issue, namespacing can be temporarily disabled:
+
+        ```julia
+        @named sys = MyModel()
+        sys = toggle_namespacing(sys, false)
+        inputs = [sys.input_var]
+        sys = toggle_namespacing(sys, true)
+        mtkcompile(sys; inputs)
```

Comment:
```suggestion
        sys_nns = toggle_namespacing(sys, false)
        inputs = [sys_nns.input_var]
        mtkcompile(sys; inputs)
```
would this work? It would save one line and does not make the code stateful